sql/src/main/java/io/crate/expression/udf/UserDefinedFunctionService.java

```diff
@@ -195,22 +199,49 @@ UserDefinedFunctionsMetaData removeFunction(@Nullable UserDefinedFunctionsMetaDa
         }
     }
 
-    public void updateImplementations(String schema, Stream<UserDefinedFunctionMetaData> userDefinedFunctions) {
-        functions.registerUdfResolversForSchema(schema, constructScalarInstances(userDefinedFunctions));
-    }
 
-    private Map<FunctionIdent, FunctionImplementation> constructScalarInstances(Stream<UserDefinedFunctionMetaData> functions) {
-        Iterator<UserDefinedFunctionMetaData> it = functions.iterator();
-        Map<FunctionIdent, FunctionImplementation> implementationsByIdent = new HashMap<>();
+    public void updateImplementations(String schema, Stream<UserDefinedFunctionMetaData> userDefinedFunctions) {
+        final Map<FunctionName, List<FuncResolver>> implementations = new HashMap<>();
+        Iterator<UserDefinedFunctionMetaData> it = userDefinedFunctions.iterator();
         while (it.hasNext()) {
-            UserDefinedFunctionMetaData udfMetaData = it.next();
-            try {
-                Scalar scalar = getLanguage(udfMetaData.language()).createFunctionImplementation(udfMetaData);
-                implementationsByIdent.put(scalar.info().ident(), scalar);
-            } catch (ScriptException | IllegalArgumentException e) {
-                LOGGER.warn("Can't create user defined function: " + udfMetaData.specificName(), e);
+            UserDefinedFunctionMetaData udf = it.next();
+            FuncResolver resolver = buildFunctionResolver(udf);
+            if (resolver == null) {
+                continue;
             }
+            implementations.compute(
+                new FunctionName(udf.schema(), udf.name()),
+                (key, resolvers) -> {
+                    if (resolvers == null) {
+                        resolvers = new ArrayList<>();
+                    }
+                    resolvers.add(resolver);
+                    return resolvers;
+                });
```

Comment:
I'd prefer using this style for improved readability, but maybe thats more a personal preference:

```
List<FuncResolver> resolvers = implementations.computeIfAbsent(functionName, new ArrayList<>());
resolvers.add(udfService.buildFunctionResolver(udf));
```

Reply:
I did it intentionally, with the reasoning that the `compute` is an atomic operation, etc. but actually it makes no sense such as everything anyways happens in the local scope of the method `UserDefinedFunctionService#updateImplementations`. I am fine with updating it to your suggestion as well.